src/services/expenses.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+let expenses = [];
+
+function emptyExpenses() {
+  expenses = [];
+
+  return expenses;
+}
+
+function getAll({ userId, category, from, to }) {
+  return expenses.length
+    ? expenses.filter(expense => {
+      const isUserIdMatch = userId
+        ? expense.userId === +userId
+        : true;
+
+      const isCategoryMatch = category
+        ? expense.category === category
+        : true;
+
+      const isFromMatch = from
+        ? expense.spentAt >= from
+        : true;
+
+      const isToMatch = to
+        ? expense.spentAt <= to
+        : true;
+
+      return isUserIdMatch && isCategoryMatch && isFromMatch && isToMatch;
+    })
+    : [];
+}
+
+function findById(expenseId) {
+  const foundExpense = expenses.find(
+    expense => expense.id === Number(expenseId)
+  );
+
+  return foundExpense || null;
+}
+
+function create(expenseData) {
+  const maxId = expenses.length
+    ? Math.max(...expenses.map(user => user.id)) + 1
+    : 0;
```

Comment:
- It's not maxId because you + 1 to Math.max(...).
- Rename variable to newId and replace 0 to 1 or remove + 1

```suggestion
  const maxId = expenses.length
    ? Math.max(...expenses.map(user => user.id)) + 1 //<<<===
    : 0;
```

Reply:
fixed